api/python/provisioner/commands/check.py

```diff
@@ -609,6 +679,239 @@ def _cluster_status(*, args: str) -> CheckEntry:
+                    check_ent.set_fail(checked_target=local_minion_id(),
+                                       comment=str(exc))
+                else:
+                    check_ent.set_passed(checked_target=hostname)
+                res.append(check_ent)
+
+        except Exception as exc:
+            check_salt.set_fail(checked_target=local_minion_id(),
+                                comment=str(exc))
+            res.append(check_salt)
+        return res
+
+    @staticmethod
+    def _public_data_ip(*, args: str) -> List[CheckEntry]:
+        """
+        Check public data ip
+
+        :param args: public_data_ip check specific parameters and arguments
+        :return:
+        """
+        _IP_DEV_IFACE_TMPL = "ip addr show dev {ifc} | grep 'inet' | grep '{ifc}'"
+        res: List[CheckEntry] = list()
+
+        # Check for import errors
+        if cortx_py_utils_import_error:
+            _res: CheckEntry = CheckEntry(cfg.Checks.PUB_DATA_IP.value)
+            _res.set_fail(checked_target=cfg.ALL_MINIONS,
+                          comment="Package cortx-py-utils not installed")
+            res.append(_res)
+            return res
+
+        check_salt: CheckEntry = CheckEntry(cfg.Checks.PUB_DATA_IP.value)
+
+        try:
+            # Get node list
+            nodes = Check._get_pillar_data("cluster/node_list")
+            for node in nodes:
+                check_ret: CheckEntry = CheckEntry(cfg.Checks.PUB_DATA_IP.value)
+
+                try:
+                    # Get list data network interfaces
+                    iface = Check._get_pillar_data(f"cluster/{node}/network/data_nw/iface")
+
+                    ifc = [ifc for ifc in iface if 's0f0' in ifc]
+
+                    if ifc:
+                        # Get IP address assigned to interface
+                        cmd = _IP_DEV_IFACE_TMPL.format(ifc=ifc[0])
+                        _ip = cmd_run(cmd, targets=node,
+                                      fun_kwargs=dict(python_shell=True))
+
+                        # Extract IP address
+                        ip = _ip[node].split()[1].split('/')[0]
+
+                        # Check for connectivity of received IP
+                        NetworkV().validate('connectivity', [ip])
+                        check_ret.set_passed(checked_target=node)
+                    else:
+                        check_ret.set_fail(checked_target=node,
+                                           comment="Public data interface " \
+                                                    "'s0f0' not found. iface "\
+                                                    f"available: {iface}")
+
+                except Exception as exc:
+                    check_ret.set_fail(checked_target=node, comment=str(exc))
+
+                res.append(check_ret)
+
+        except Exception as exc:
+            check_salt.set_fail(checked_target=cfg.ALL_MINIONS, comment=str(exc))
+            res.append(check_salt)
+
+        return res
```

Comment:
It can be simplified for more readability:
```python
@staticmethod
def _public_data_ip(*, args: str) -> Union[CheckEntry, List[CheckEntry]]:
    """
    Check public data ip
    :param args: public_data_ip check specific parameters and arguments
    :return:
    """
    _IP_DEV_IFACE_TMPL = "ip addr show dev {ifc} | grep 'inet' | grep '{ifc}'"
    check_entry: CheckEntry = CheckEntry(cfg.Checks.PUB_DATA_IP.value)

    # Check for import errors
    if cortx_py_utils_import_error:
        check_entry.set_fail(checked_target=cfg.ALL_MINIONS,
                      comment="Package cortx-py-utils not installed")
        return check_entry

    try:
        # Get node list
        nodes = Check._get_pillar_data("cluster/node_list")
    except Exception as exc:
        check_entry.set_fail(checked_target=cfg.ALL_MINIONS, comment=str(exc))
        return check_entry

    res: List[CheckEntry] = list()
    for node in nodes:
        check_ret: CheckEntry = CheckEntry(cfg.Checks.PUB_DATA_IP.value)

        try:
            # Get list data network interfaces
            iface = Check._get_pillar_data(
                f"cluster/{node}/network/data_nw/iface")

            ifc = [ifc for ifc in iface if 's0f0' in ifc]

            if ifc:
                # Get IP address assigned to interface
                cmd = _IP_DEV_IFACE_TMPL.format(ifc=ifc[0])
                _ip = cmd_run(cmd, targets=node,
                              fun_kwargs=dict(python_shell=True))

                # Extract IP address
                ip = _ip[node].split()[1].split('/')[0]

                # Check for connectivity of received IP
                NetworkV().validate('connectivity', [ip])
                check_ret.set_passed(checked_target=node)
            else:
                check_ret.set_fail(checked_target=node,
                                   comment="Public data interface " \
                                           "'s0f0' not found. iface " \
                                           f"available: {iface}")
        except Exception as exc:
            check_ret.set_fail(checked_target=node, comment=str(exc))

        res.append(check_ret)

    return res
```

Reply:
Done